CMakeLists.txt

```diff
@@ -0,0 +1,117 @@
+cmake_minimum_required(VERSION 2.8.12)
+
+project(cctz)
+
+set(CMAKE_MODULE_PATH
+  ${PROJECT_SOURCE_DIR}/cmake
+  ${PROJECT_SOURCE_DIR}/cmake/modules
+  ${CMAKE_MODULE_PATH})
+
+include(CheckCXXCompilerFlag)
+include(CTest)
+include(GNUInstallDirs)
+
+option(BUILD_EXAMPLES "Whether or not to build examples" ON)
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+  add_compile_options(-Wall)
+endif()
+
+function(cctz_target_set_cxx_standard target)
+  set(cctz_cxx_standard 11)
+  # CXX_STANDARD target property is only supported since CMake >= 3.1
+  if (CMAKE_VERSION VERSION_LESS "3.1")
+    if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+      target_compile_options(${target} PRIVATE -std=c++${cctz_cxx_standard})
+    else()
+      # if a specific standard is required, this can be set from the command line:
+      #     cmake -DCMAKE_CXX_STANDARD=[11|14|17]
+      set_property(TARGET ${target} PROPERTY CXX_STANDARD ${cctz_cxx_standard})
+    endif()
+  endif()
+endfunction()
+
+add_library(cctz
+  src/civil_time_detail.cc
+  src/time_zone_fixed.cc
+  src/time_zone_fixed.h
+  src/time_zone_format.cc
+  src/time_zone_if.cc
+  src/time_zone_if.h
+  src/time_zone_impl.cc
+  src/time_zone_impl.h
+  src/time_zone_info.cc
+  src/time_zone_info.h
+  src/time_zone_libc.cc
+  src/time_zone_libc.h
+  src/time_zone_lookup.cc
+  src/time_zone_posix.cc
+  src/time_zone_posix.h
+  src/tzfile.h
+  src/zone_info_source.cc
+  )
+cctz_target_set_cxx_standard(cctz)
+add_library(cctz::cctz ALIAS cctz)
+
+target_include_directories(cctz PUBLIC PUBLIC include)
+
+# not (yet) supported on Windows because of getopt.h
```

Comment:
Use CHECK_INCLUDE_FILE instead of MSVC?

Reply:
Done with `check_cxx_symbol_exists`, I assume it's theoretically possible for a system to have getopt.h without getopt_long, which is a GNU extension.
The CMake code is slightly more complicated with this change, but I guess the right thing to do would be to not use `getopt_long`.